crates/astria-sequencer/src/metrics.rs

```diff
@@ -212,6 +292,59 @@ impl Metrics {
     pub(crate) fn increment_check_tx_removed_account_balance(&self) {
         self.check_tx_removed_account_balance.increment(1);
     }
+
+    pub(crate) fn record_check_tx_duration_seconds_parse_tx(&self, duration: Duration) {
+        self.check_tx_duration_seconds_parse_tx.record(duration);
+    }
+
+    pub(crate) fn record_check_tx_duration_seconds_check_stateless(&self, duration: Duration) {
+        self.check_tx_duration_seconds_check_stateless
+            .record(duration);
+    }
+
+    pub(crate) fn record_check_tx_duration_seconds_check_nonce(&self, duration: Duration) {
+        self.check_tx_duration_seconds_check_nonce.record(duration);
+    }
+
+    pub(crate) fn record_check_tx_duration_seconds_check_chain_id(&self, duration: Duration) {
+        self.check_tx_duration_seconds_check_chain_id
+            .record(duration);
+    }
+
+    pub(crate) fn record_check_tx_duration_seconds_check_balance(&self, duration: Duration) {
+        self.check_tx_duration_seconds_check_balance
+            .record(duration);
+    }
+
+    pub(crate) fn record_check_tx_duration_seconds_check_removed(&self, duration: Duration) {
+        self.check_tx_duration_seconds_check_removed
+            .record(duration);
+    }
+
+    pub(crate) fn record_check_tx_duration_seconds_insert_to_app_mempool(
+        &self,
+        duration: Duration,
+    ) {
+        self.check_tx_duration_seconds_insert_to_app_mempool
+            .record(duration);
+    }
+
+    pub(crate) fn record_actions_per_transaction_in_mempool(&self, count: usize) {
+        // allow: precision loss is unlikely (values too small) but also unimportant in histograms.
+        #[allow(clippy::cast_precision_loss)]
+        self.actions_per_transaction_in_mempool.record(count as f64);
+    }
+
+    pub(crate) fn record_transaction_in_mempool_size_bytes(&self, count: usize) {
+        // allow: precision loss is unlikely (values too small) but also unimportant in histograms.
+        #[allow(clippy::cast_precision_loss)]
+        self.transaction_in_mempool_size_bytes.record(count as f64);
+    }
+
```

Comment:
these are set every time a tx is inserted (and is per-tx) right? would it be more/less useful to also have a mempool total?